stock_move_free_reservation_reassign/models/stock_move_line.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from contextlib import contextmanager
+
+from odoo import models
+
+
+class StockMoveLine(models.Model):
+
+    _inherit = "stock.move.line"
+
+    @contextmanager
+    def _get_move_free_reservation_ids(self):
+        """A context manager method to collect the move where the reservation has
+        been freed
+
+        It will wrap the `_free_reservation` method of the `stock.move.line` model.
+        to collect the move impacted by the reservation removal.
+        """
+        wrapped = self.env["stock.move.line"].__class__._free_reservation
+        move_to_reassign_ids = set()
+
+        def _free_reservation(*args, **kwargs):
+            with args[0]._get_move_recomputed_state_ids() as move_recomputed_state_ids:
+                res = wrapped(*args, **kwargs)
+                move_to_reassign_ids.update(move_recomputed_state_ids)
+                return res
+
+        try:
```

Comment:
I'm wondering what would happen if, while `stock.quant._apply_inventory` is being executed, `stock.move.line.free_reservation` is called.